File: sach/de3342.py
//DE3342.PY - Tran Huu Nam - hsgtin.vn - 2024
import math
def U(n):
    dem = 0
    cn = int(math.sqrt(n))
    if cn * cn == n:
        dem += 1
        cn -= 1
    for i in range(1, cn + 1):
        if n % i == 0:
            dem += 2
    return dem
def main():
    n = int(input())
    res = 0
    for _ in range(n):
        x = int(input())
        res += x * U(x)
    print(res)
if __name__ == "__main__":
    main()

File: sach/de3342.cpp
//DE3342.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
typedef long long ll;
using namespace std;
int U(int n){
    int dem=0, cn = sqrt(n);
    if (cn*cn==n) {
        dem+=1;
        cn--;
    }
    for(int i=1; i<=cn; i++)
        if(n%i==0) dem+=2;
    return dem;
}
int main() {
    int n, x;     ll res=0;
    cin >> n;
    for(int i=1; i<=n; i++){
        cin >> x;
        res+=x*U(x);
    }
    cout << res;
    return 0;
}

